### PR TITLE
Add storyshots

### PR DESCRIPTION
**Pull request by Tanuj22**
**Fixes**

Fixes #165 by @dhruvkb 

**Description**
Add Shoryshots testing 

**Type of PR**

This PR is a feature.

**Technicalities**

**Tests**

**Screenshots**

**Checklist:**
 
- [ ] My pull request has a descriptive title (not a vague title like "Update `index.md`").
- [ ] My pull request targets the `master` branch of the repository.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

**Legal**
By submitting this PR, I agree to abide by the terms of the Developer 
Certificate of Origin.

<details>
<summary>Developer Certificate of Origin</summary>

Developer Certificate of Origin<br>
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.<br>
1 Letterman Drive<br>
Suite D4700<br>
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
</details>

### CODE REVIEW

**Comment by Tanuj22** [comment]
@dhruvkb I have taken the reference of [chooser](https://github.com/creativecommons/chooser) repo to configure Jest for ES6 syntax but getting the following error. What seems to cause the error?
Screenshot from 2020-03-19 15-51-27

File: package.json [hunk 1]
```diff
@@ -5,18 +5,22 @@
   "author": "Creative Commons (https://creativecommons.org)",
   "scripts": {
     "lint": "eslint src/*.js src/**/*.js",
-    "test": "jest tests/sass.test.js",
+    "test": "jest",
```
Comment (dhruvkb) [inline package.json]: The name "test" is already taken for running True tests. You could use a name like `shots` or similar.

File: package.json [hunk 2]
```diff
@@ -54,5 +58,18 @@
...
+      "json",
+      "scss"
+    ],
+    "moduleNameMapper": {
+      "^@/(.*)$": "<rootDir>/src/$1"
+    },
+    "transform": {
+      "^.+\\.js$": "<rootDir>/node_modules/babel-jest"
+    }
+  }
```
Comment (dhruvkb) [inline package.json]: Would prefer configurations to not be in `package.json` and rather be in a separate file.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
Is the PR taking any shots at the moment?

File: package.json (hunk 1 above)
Comment (dhruvkb) [reply package.json]: Or if you could run Storyshots with the same command (while not disturbing the current setup with True), that would be cool too.

**Comment by Tanuj22** [comment]
It's not taking any shots currently. After adding `.babelrc` it was able to recognize ES6 syntax in `storybook.test.js` but not in other files. Can't seem to figure out why.

File: package.json (hunk 2 above)
Comment (Tanuj22) [reply package.json]: Okay

File: package.json (hunk 1 above)
Comment (Tanuj22) [reply package.json]: Doing this runs all the files with `test.js` extension.

**Comment by dhruvkb** [comment]
@Tanuj22 with the merge of #227, we got `import` statements working (to some extent). Update your branch to see if you can make it work now.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
Retry with PR #227 merged.

**Comment by Tanuj22** [comment]
@dhruvkb I tried again but still the same error. I took the reference of how its setup in one of the [example](https://github.com/storybookjs/storybook/tree/master/examples/html-kitchen-sink) in Storybook repo. Can't figure out whats missing.

**Comment by Tanuj22** [comment]
I was able to fix the issue with import statements. Now stuck with this error :/
Screenshot from 2020-04-01 20-11-53

**Comment by Tanuj22** [comment]
@dhruvkb Was finally able to make it work. Please have a look

**Review by dhruvkb — COMMENTED** [review]
Why do the snapshot files have so many blank lines?

**Comment by Tanuj22** [comment]
Not really sure. I guess it may have something to do with converting the MDX files by `@storybook/addon-docs/jest-transform-mdx`

**Comment by dhruvkb** [comment]
@Tanuj22 can we do something about it? A lot of the files are just blanks. Doesn't make much sense to have them then.

**Comment by Tanuj22** [comment]
@dhruvkb I was also disappointed by the results. I will try to find any solution and get back to you.

**Comment by Tanuj22** [comment]
@dhruvkb I can't find a solution to this. Maybe it's best to close the PR.

**Comment by dhruvkb** [comment]
Let's not close it but rather get some help from the community.

**Comment by dhruvkb** [comment]
This PR has been going stale from inactivity.

**Pull request closed without merging**